rclcpp/src/rclcpp/executors/multi_threaded_executor.cpp

```diff
@@ -99,6 +99,19 @@ MultiThreadedExecutor::run(size_t this_thread_number)
 
     execute_any_executable(any_exec);
 
+    if (any_exec.callback_group &&
+      any_exec.callback_group->type() == CallbackGroupType::MutuallyExclusive &&
+      any_exec.callback_group->size() > 1)
```

Comment:
Since this guard condition is the thing that will wake up the WaitSet in another thread, I think it only makes sense to trigger in the case that there is actually work available.

In my mind, in the case that there is a callback group with only one thing in it, then triggering the guard condition doesn't really accomplish anything other than a spurious wakeup of the waitset.

Reply:
Add a comment ?

Reply:
After thinking about this a little more, I'm going to remove the condition because it doesn't correctly account for group type changing or group membership changing during the course of the executable entity being run.

Reply:
Sounds good, I'm still kind of confused about what this condition was checking though, because a `> 0` kind of makes sense, as in "no need to notify if the guard condition is empty", but `> 1` didn't really track for me.